projects/project5/Group5/src/main/scala/ds/TripleReader.scala

```diff
@@ -0,0 +1,223 @@
+
+package ds
+import java.net.URI
+
+import scala.collection.mutable.HashMap
+
+import net.sansa_stack.rdf.spark.io._
+
+import net.sansa_stack.rdf.spark.model._
+
+import org.apache.spark.sql.SparkSession
+
+import org.apache.jena.riot.Lang
+
+import scala.collection.mutable
+
+import org.apache.jena.graph.NodeFactory
+
+import scala.collection.mutable.Map
+
+import scala.collection.mutable.ListBuffer
+import ds._
+
+
+object TripleReader {
+
+
+
+  def main(args: Array[String]) {
+
+    parser.parse(args, Config()) match {
+
+      case Some(config) =>
+
+        run(config.in)
+
+      case None =>
+
+        println(parser.usage)
+
+    }
+
+  }
+
+
+
+  def run(input: String): Unit = {
+
+
+
+    val spark = SparkSession.builder
+
+      .appName(s"Triple reader example  $input")
+
+      .master("local[*]")
+
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+
+      .getOrCreate()
+
+
+
+    println("======================================")
+
+    println("|        Triple reader example       |")
+
+    println("======================================")
+
+
+
+    val lang = Lang.NTRIPLES
+
+
+
+    val triples = spark.rdf(lang)(input)
+
+    var dict:Map[String,String] = Map()
+
+    var atomlist = triples.collect().toList
```

Comment:
From here you are doing everything on the List/Arrays level, not sure why Spark adv. hasn't been considered afterward.  If the size of the dataset is big, how will you handle it?